exporter/datadogexporter/traces_exporter.go

```diff
@@ -16,25 +16,91 @@ package datadogexporter
 
 import (
 	"context"
+	"fmt"
 
+	"github.com/DataDog/datadog-agent/pkg/trace/obfuscate"
 	"go.opentelemetry.io/collector/consumer/pdata"
 	"go.uber.org/zap"
 )
 
 type traceExporter struct {
-	logger *zap.Logger
+	logger         *zap.Logger
+	cfg            *Config
+	edgeConnection TraceEdgeConnection
+	obfuscator     *obfuscate.Obfuscator
+	tags           []string
 }
 
-func newTraceExporter(_ *Config, logger *zap.Logger) (*traceExporter, error) {
-	return &traceExporter{logger}, nil
+func newTraceExporter(logger *zap.Logger, cfg *Config) (*traceExporter, error) {
+	// removes potentially sensitive info and PII, approach taken from serverless approach
+	// https://github.com/DataDog/datadog-serverless-functions/blob/11f170eac105d66be30f18eda09eca791bc0d31b/aws/logs_monitoring/trace_forwarder/cmd/trace/main.go#L43
+	obfuscator := obfuscate.NewObfuscator(&obfuscate.Config{
+		ES: obfuscate.JSONSettings{
+			Enabled: true,
+		},
+		Mongo: obfuscate.JSONSettings{
+			Enabled: true,
+		},
+		RemoveQueryString: true,
+		RemovePathDigits:  true,
+		RemoveStackTraces: true,
+		Redis:             true,
+		Memcached:         true,
+	})
+
+	// Calculate tags at startup
+	tags := cfg.TagsConfig.GetTags(false)
+	// TODO:
+	// use passed in config values for site and api key instead of hardcoded
+	exporter := &traceExporter{
+		logger:         logger,
+		cfg:            cfg,
+		edgeConnection: CreateTraceEdgeConnection(cfg.Traces.TCPAddr.Endpoint, cfg.API.Key, false),
+		obfuscator:     obfuscator,
+		tags:           tags,
+	}
+
+	return exporter, nil
 }
 
 func (exp *traceExporter) pushTraceData(
-	_ context.Context,
+	ctx context.Context,
 	td pdata.Traces,
 ) (int, error) {
 
-	// TODO: add implementation
+	// convert traces to datadog traces and group trace payloads by env
+	// we largely apply the same logic as the serverless implementation, simplified a bit
+	// https://github.com/DataDog/datadog-serverless-functions/blob/f5c3aedfec5ba223b11b76a4239fcbf35ec7d045/aws/logs_monitoring/trace_forwarder/cmd/trace/main.go#L61-L83
+	ddTraces, err := convertToDatadogTd(td, exp.cfg, exp.tags)
+
+	if err != nil {
+		exp.logger.Info(fmt.Sprintf("Failed to convert traces with error %v\n", err))
+		return 0, err
+	}
+
+	// group the traces by env to reduce the number of flushes
+	aggregatedTraces := aggregateTracePayloadsByEnv(ddTraces)
+
+	// security/obfuscation for db, query strings, stack traces, pii, etc
+	// TODO: is there any config we want here? OTEL has their own pipeline for regex obfuscation
```

Comment:
There's some span processors that allow regex to modify/filter specific attribs. 

- https://github.com/open-telemetry/opentelemetry-collector/tree/master/processor/attributesprocessor
- https://github.com/open-telemetry/opentelemetry-collector/tree/master/processor/spanprocessor

It's independent from a datadog exporters but i think it would be nice to eventually have documentation of suggested best practices for pipeline setups, or even plug n' play yaml examples for different use cases that we could have live in either a datadog exporter specific readme or docs section.